### DISCUSSION TITLE
Rule for: `prefer object?.prop` over `object && object.prop` exisits?

### DISCUSSION THREAD

**Discussion by pke** [eslint/eslint#15068] [Q&A] [answered]
Is there already a rule that catches `object && object.prop` and suggest to change it to `object?.prop`?

**Comment by nzakas** [comment]
No there isn’t. You can see all rules at https://eslint.org/docs/rules

**Reply by pke** [reply]
Would a PR for that make sense or could such linter suggestion lead to bugs in the code?

**Reply by mdjermanovic** [reply, marked as answer]
There is a proposal for a rule that would suggest `object?.prop`: https://github.com/eslint/eslint/issues/13430

The proposal is generally accepted, but it seems that it needs more design. In particular, `object && object.prop` might be problematic for the reasons mentioned in the comments on the issue.

**Reply by pke** [reply]
Thanks for the hint. We will discuss over there more.
How can one mark your comment as answer? ;)
